src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -385,6 +397,52 @@ JNI_METHOD(jlong, newDeviceController)(JNIEnv * env, jobject self, jobject contr
                 SuccessOrExit(err);
             }
         }
+
+        jobject countryCode;
+        err = chip::JniReferences::GetInstance().GetOptionalValue(countryCodeOptional, countryCode);
+        SuccessOrExit(err);
+
+        if (countryCode != nullptr)
+        {
+            jstring countryCodeStr = static_cast<jstring>(countryCode);
+            JniUtfString countryCodeJniString(env, countryCodeStr);
+
+            VerifyOrExit(countryCodeJniString.size() == 2, err = CHIP_ERROR_INVALID_ARGUMENT);
+
+            chip::Controller::CommissioningParameters commissioningParams = wrapper->GetCommissioningParameters();
+            commissioningParams.SetCountryCode(countryCodeJniString.charSpan());
```

Comment:
>     // The lifetime of the buffer countryCode is pointing to should exceed the
>     // lifetime of CommissioningParameters object.

^^^ You must add a member in the wrapper that is a char[2], and copy from the Java object to that storage, and then call SetCountryCode. Otherwise, the countryCodeJniString string will go out of scope and the span will point to garbage.